packages/components/src/snackbar/index.tsx

```diff
@@ -138,9 +138,12 @@ function UnforwardedSnackbar(
 			className={ classes }
 			onClick={ ! explicitDismiss ? dismissMe : undefined }
 			tabIndex={ 0 }
-			role={ ! explicitDismiss ? 'button' : '' }
+			role={ ! explicitDismiss ? 'button' : undefined }
 			onKeyPress={ ! explicitDismiss ? dismissMe : undefined }
-			aria-label={ ! explicitDismiss ? __( 'Dismiss this notice' ) : '' }
+			aria-label={
+				! explicitDismiss ? __( 'Dismiss this notice' ) : undefined
+			}
```

Comment:
Prevents attributes without values from being applied.

Reply:
The negation in all of these ternaries feels totally backwards to me, but hey ho... at least they're consistent!

Perhaps instead we could separate out the dismiss props?

```jsx
// `explicitDismiss || { ... }` also works,
// as long as it's guaranteed to be a boolean!

const dismissProps = explicitDismiss ? {} : {
  onClick: dismissMe,
  onKeyPress: dismissMe,
  role: 'button',
  'aria-label': __( 'Dismiss this notice' ),
}

return (
  <div { ...dismissProps } />
);
```

Anyway, scope creep 🚨

Reply:
Although this is a negative form, how about the following approach?

```javascript
const dismissProps = {
	onClick: dismissMe,
	onKeyPress: dismissMe,
	role: 'button',
	'aria-label': __( 'Dismiss this notice' ),
};

return (
	<div
		// When there is no explicit close button to close the snackbar,
		// give the snackbar itself a prop that can be dismissed.
		{ ...( ! explicitDismiss && dismissProps ) }
	>
);
```